src/controllers/legacycontrollersettings.cpp

```diff
@@ -0,0 +1,213 @@
+                                    disposition) {
+                        pLayout->setDirection(disposition ==
+                                                LegacyControllerSettingsLayoutContainer::
+                                                        HORIZONTAL
+                                        ? QBoxLayout::TopToBottom
+                                        : QBoxLayout::LeftToRight);
+                        pParent->layout()->invalidate();
+                    });
+        }
+    }
+
+    auto pLabelWidget = make_parented<QLabel>(pRoot);
+    pLabelWidget->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Maximum);
+    pLabelWidget->setText(label());
+
+    if (!description().isEmpty()) {
+        pRoot->setToolTip(QString("<p>%1</p>").arg(description()));
+    }
+
+    pLayout->addWidget(pLabelWidget);
+    pLayout->addWidget(buildInputWidget(pRoot));
+
+    pLayout->setStretch(0, 3);
+    pLayout->setStretch(1, 1);
+
+    pRoot->setLayout(pLayout);
+
+    return pRoot;
+}
+
+LegacyControllerBooleanSetting::LegacyControllerBooleanSetting(
+        const QDomElement& element)
+        : AbstractLegacyControllerSetting(element) {
+    m_defaultValue = parseValue(element.attribute("default"));
+    m_savedValue = m_defaultValue;
+    m_editedValue = m_defaultValue;
+}
+
+QWidget* LegacyControllerBooleanSetting::buildWidget(
+        QWidget* pParent, LegacyControllerSettingsLayoutContainer::Disposition) {
+    return buildInputWidget(pParent);
+}
+
+QWidget* LegacyControllerBooleanSetting::buildInputWidget(QWidget* pParent) {
+    auto pCheckBox = make_parented<QCheckBox>(label(), pParent);
+    pCheckBox->setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Fixed);
+    if (m_savedValue) {
+        pCheckBox->setCheckState(Qt::Checked);
+    }
+
+    if (!description().isEmpty()) {
+        pCheckBox->setToolTip(QString("<p>%1</p>").arg(description()));
+    }
+
+    connect(pCheckBox, &QCheckBox::stateChanged, this, [this](int state) {
+        m_editedValue = state == Qt::Checked;
+        emit changed();
+    });
+
+    return pCheckBox;
+}
+
+bool LegacyControllerBooleanSetting::match(const QDomElement& element) {
+    return element.hasAttribute("type") &&
+            QString::compare(element.attribute("type"),
+                    "boolean",
+                    Qt::CaseInsensitive) == 0;
+}
+
+REGISTER_LEGACY_CONTROLLER_SETTING(LegacyControllerBooleanSetting);
+
+template<class SettingType,
+        Serializer<SettingType> ValueSerializer,
+        Deserializer<SettingType> ValueDeserializer,
+        class InputWidget>
+QWidget* LegacyControllerNumberSetting<SettingType,
+        ValueSerializer,
+        ValueDeserializer,
+        InputWidget>::buildInputWidget(QWidget* pParent) {
+    auto spinBox = make_parented<InputWidget>(pParent);
+    spinBox->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Maximum);
+
+    spinBox->setRange(this->m_minValue, this->m_maxValue);
```

Comment:
In C++ this-> is redundant 
```suggestion
    spinBox->setRange(m_minValue, m_maxValue);
```